ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,896 @@
+
+            rows.append(r)
+
+        return rows
+
+    @property
+    def n_dims(self):
+        """The dimensionality of the original space."""
+        return len(self.dimensions)
+
+    @property
+    def transformed_n_dims(self):
+        """The dimensionality of the warped space."""
+        return sum([dim.transformed_size for dim in self.dimensions])
+
+    @property
+    def bounds(self):
+        """The dimension bounds, in the original space."""
+        b = []
+
+        for dim in self.dimensions:
+            if dim.size == 1:
+                b.append(dim.bounds)
+            else:
+                b.extend(dim.bounds)
+
+        return b
+    
+    @property
+    def names(self):
+        '''The names of the dimensions if given. Otherwise [X1, X2, ... Xn]'''
+        labels = ["$X_{%i}$" % i if d.name is None else d.name
+            for i, d in enumerate(self.dimensions)]
+        return labels
+
+    def __contains__(self, point):
+        """Check that `point` is within the bounds of the space."""
+        for component, dim in zip(point, self.dimensions):
+            if component not in dim:
+                return False
+        return True
+
+    @property
+    def transformed_bounds(self):
+        """The dimension bounds, in the warped space."""
+        b = []
+
+        for dim in self.dimensions:
+            if dim.transformed_size == 1:
+                b.append(dim.transformed_bounds)
+            else:
+                b.extend(dim.transformed_bounds)
+
+        return b
+
+    @property
+    def is_categorical(self):
+        """Space contains exclusively categorical dimensions"""
+        return all([isinstance(dim, Categorical) for dim in self.dimensions])
+
+    @property
+    def is_partly_categorical(self):
+        """Space contains any categorical dimensions"""
+        return any([isinstance(dim, Categorical) for dim in self.dimensions])
+
+    def distance(self, point_a, point_b):
+        """Compute distance between two points in this space.
+
+        Parameters
+        ----------
+        * `a` [array]
+            First point.
+
+        * `b` [array]
+            Second point.
+        """
+        distance = 0.
+        if len(self.dimensions) > 1:
+            for a, b, dim in zip(point_a, point_b, self.dimensions):
+                distance += dim.distance(a, b)
+                
+        if len(self.dimensions) == 1:
+             distance +=  self.dimensions[0].distance(point_a[0], point_b[0])
+
+        return distance
+
+    def lhs(self, n):
+        """ Returns n latin hypercube samples as a list of lists
+        """
+
+        samples = []
+        for i in range(self.n_dims):
+            lhs_perm = []
+            # Get evenly distributed samples from one dimension
+            sample_indices = (np.arange(n) + 0.5) / n
+            lhs_aranged = self.dimensions[i].sample(sample_indices)
+            perm = np.random.RandomState(seed=42 + i).permutation(n)
```

Comment:
While we are at it, I suggest changing this line to use np.random.default_rng with a set seed instead.

Reply:
I had completely missed that use of RandomState.

We might as well open up for actually having LHS being random, for consistency.

Reply:
It would certainly help benchmarking, since right now you always get the exact same LHS set of points, no matter what. This makes your results very system-specific, as you always start optimizations from the same initial points, which may happen to be really good (or bad) for the given "landscape".

Reply:
Fixed.